index.js

```diff
@@ -88,27 +104,54 @@ class CSVPlugin {
     return this.context.json.expand(data)
   }
 
+  loadExportTemplates(data) {
+    const state = this.context.window.store?.getState()
+    const inferredItemTemplate = data['@graph']?.[0]?.template
+    const itemTemplate = loadTemplate(
+      state,
+      this.options.itemTemplate,
+      inferredItemTemplate)
+
+    if (!itemTemplate)
+      // eslint-disable-next-line max-len
+      throw new Error(`Failed to find specified item template "${this.options.itemTemplate}" or fallback "${data['@graph']?.[0]?.template}". Please install one of these templates, or select a different template and try again.`)
+
+
+    const inferredPhotoTemplate = data['@graph']?.[0]?.photo?.[0]?.template
+    const photoTemplate = this.options.photoMetadata ? loadTemplate(
+      state,
+      this.options.photoTemplate,
+      inferredPhotoTemplate) : null
+
+    if (this.options.photoMetadata && !photoTemplate)
+      // eslint-disable-next-line max-len
+      throw new Error(`Failed to find specified photo template "${this.options.photoTemplate}" or fallback "${data['@graph']?.[0]?.photo?.[0]?.template}". Please install one of these templates, or select a different template and try again.`)
+
+    return { itemTemplate, photoTemplate }
+  }
+
+  maxPhotos(data) {
```

Comment:
Did I make the correct assumptions here about what the shape of the input data can be?

Reply:
The data is json-ld and I'm pretty sure the way Tropy hands it to the plugin will always be as an array of photo objects (not using the expanded `@list` syntax). I guess that's what you're asking, right? So I'll say yes, it's correct, but it's an important assumption (like elsewhere when we access non-expanded data).

Reply:
Is it worth documenting this assumption in the function? If so, jsdoc comments or some other format?

Reply:
It's never wrong to document assumptions (just regular code comments), but if you want to add a note I'd add it at the export function itself, since the plugin relies on the data coming from Tropy. In practice, we'll have to check and update all plugins if we make a change to the data that Tropy hands to the plugins. To be safe, I think a plugin would absolutely have to use the json-ld library to convert the data to a canonical form before using -- but this is no fun either.